jsconfig.json

```diff
@@ -0,0 +1,19 @@
+{
+  "compilerOptions": {
+    "target": "es5",
+    "module": "esnext",
+    "baseUrl": "./",
+    "moduleResolution": "node",
+    "paths": {
+      "@/*": ["src/*"],
+      "APP/*": ["app/*"]
```

Comment:
Could you give me some insights why the alias paths go in here, since this appears to be partially redundant to what we're doing in `vue.config.js::chainWebpack`?  I noticed this code is part of the cli template, so theres nothing wrong with it. I'm curious because I tried to remove the paths here and it if I recall correctly it didn`t have any effect. Same vice versa. My custom app introduces some other aliases, which I had to add in chainWebpack. Declaring them here did nothing for me.

Reply:
Funny that your conversations are in order of difficulty to answer here ;-) this one is not too hard to explain... well perhaps...

`jsconfig.json` is a file which at a time was specific to `VSCode` and was here to give information to its `LSP` about the project. You can find the [official documentation on VSCode here](https://code.visualstudio.com/docs/languages/jsconfig) and [the PR and discussions why it was added in Vue-CLI here](https://github.com/vuejs/vue-cli/pull/6285).
It's said there that the file was added for [Vue-CLI to work better with Vetur or VueDX](https://github.com/vuejs/vue-cli/issues/6064#issuecomment-733543301)
This file is since used by other code editors too like `SublimeText` and is here mainly to give better coding experience and provide, for example, auto-completion while writing imports.
That's why nothing happens when you modify this file, it doesn't impact the build, only the way your code editor behaves... if it makes use of such a file.

I personally use it in some projects to type hint my code with `JSDoc` and have the code editor warns me when the types are not respected (see `checkJs` in the doc) which would be very useful in `Wegue` but would take so much time to implement...

Eventually, there are commonly two things to be patched inside this file, first the alias paths for the reason given up here (autocompletion inside the code editor) and the exclude key so the code editor don't think `node_modules` is part of our project and analyses it which can take ages...
Here's [a small article](https://lmichelin.fr/vscode-intellisense-jsconfig-vue-js-nuxt-js-absolute-imports/) which says this better than I am for completeness.

Reply:
OK, thanks a lot for the explanation. And sorry in hindsight for not doing investigations on this myself :)